src/components/AddItem.js

```diff
@@ -1,7 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { writeToFirestore } from '../lib/firebase';
 
 const AddItem = () => {
-  return <h1>THIS IS THE ADD ITEM</h1>;
+  let [name, setName] = useState('');
+  let [frequency, setFrequency] = useState('');
+
+  function handleSubmitForm(event) {
+    event.preventDefault();
+    if (name.length > 0) {
+      writeToFirestore('put-token-here', {
+        name,
+        frequency,
+        lastPurchaseDate: null,
+      });
+      alert(`${name} has been successfully added to your list.`);
+      setName('');
+    }
+  }
+
+  return (
+    <form onSubmit={event => handleSubmitForm(event)}>
+      <label htmlFor="add-shopping-list-item">Item Name</label>
+      <input
```

Comment:
If you give this element a `required` attribute, I think that will automatically give you some validation.

Reply:
Added required to the first field (item name), but didn't end up adding it to the fieldset since it has a default value.